lighthouse-core/test/audits/errors-in-console-test.js

```diff
@@ -121,4 +121,36 @@ describe('Console error logs audit', () => {
     // text is undefined
     assert.strictEqual(auditResult.details.items[0][1].text, undefined);
   });
+
+  // Checks bug #4188
+  it('handle the case when exception info is not present', () => {
+    const auditResult = ErrorLogsAudit.audit({
+      ChromeConsoleMessages: [],
+      RuntimeExceptions: [{
+        'timestamp': 1506535813608.003,
+        'exceptionDetails': {
+          'url': 'http://example.com/fancybox.js',
+          'text': 'TypeError: Cannot read property \'msie\' of undefined',
+          'stackTrace': {
+            'callFrames': [
+              {
+                'url': 'http://example.com/fancybox.js',
+                'lineNumber': 28,
+                'columnNumber': 20,
+              },
+            ],
+          },
+          'executionContextId': 3,
+        },
+      }],
+    });
+    assert.equal(auditResult.rawValue, 1);
+    assert.equal(auditResult.score, false);
+    assert.equal(auditResult.details.items.length, 1);
+    // url is undefined
+    assert.strictEqual(auditResult.details.items[0][0].text, 'http://example.com/fancybox.js');
+    // text is undefined
```

Comment:
not sure what these two comments are, isn't `text` defined here?

Reply:
yes I did, it happens when a syntax error occurred on the page.

```
{
   "timestamp":1515451440948.461,
   "exceptionDetails":{
      "exceptionId":1,
      "text":"Uncaught SyntaxError: Unexpected token ,",
      "lineNumber":3180,
      "columnNumber":302,
      "scriptId":"56",
      "url":"https://enterprise.efax.com/",
      "stackTrace":{
         "callFrames":[
            {
               "functionName":"prop",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":22254
            },
            {
               "functionName":"access",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":6743
            },
            {
               "functionName":"prop",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":18120
            },
            {
               "functionName":"",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":88,
               "columnNumber":274
            },
            {
               "functionName":"each",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":5254
            },
            {
               "functionName":"each",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":2010
            },
            {
               "functionName":"b",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":88,
               "columnNumber":190
            },
            {
               "functionName":"",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":89,
               "columnNumber":472
            },
            {
               "functionName":"c",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":7854
            },
            {
               "functionName":"fireWith",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":8655
            },
            {
               "functionName":"ready",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":3263
            },
            {
               "functionName":"H",
               "scriptId":"32",
               "url":"https://d9jmv9u00p0mv.cloudfront.net/js/10015692-10015098.js",
               "lineNumber":8,
               "columnNumber":692
            }
         ]
      }
   }
}
```